### PR TITLE
feat: catch up with Kubeadm in terms of conditions

### PR DESCRIPTION
**Pull request by Unix4ever**
This is the first set of conditions support:
- `ResizedCondition`.
- `ControlPlaneComponentsHealthy`
- `EtcdClusterHealthy`
- `Available`
- `MachinesReady`
- `MachinesCreated`

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ type: kubernetes
 
 services:
   - name: docker
-    image: docker:20.10-dind
+    image: ghcr.io/smira/docker:20.10-dind-hacked
```
Comment (smira) [inline .drone.yml:8]: do we do privileged somewhere? shouldn't be needed unless we do that

File: api/v1alpha3/taloscontrolplane_types.go [hunk 2]
```diff
@@ -88,6 +89,14 @@ type TalosControlPlaneStatus struct {
 	// state, and will be set to a descriptive error message.
 	// +optional
 	FailureMessage *string `json:"failureMessage,omitempty"`
+
+	// ObservedGeneration is the latest generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```
Comment (smira) [inline api/v1alpha3/taloscontrolplane_types.go:95]: I think I need to add same to the CABPT

File: controllers/taloscontrolplane_controller.go [hunk 3]
```diff
@@ -736,77 +790,30 @@ func (r *TalosControlPlaneReconciler) addClusterOwnerToObj(ctx context.Context,
...
-			nodesNotFinishedBooting = append(nodesNotFinishedBooting, node)
-		}
-	}
-
-	sort.Strings(nodesNotFinishedBooting)
+func patchTalosControlPlane(ctx context.Context, patchHelper *patch.Helper, tcp *controlplanev1.TalosControlPlane, opts ...patch.Option) error {
+	// Always update the readyCondition by summarizing the state of other conditions.
+	conditions.SetSummary(tcp,
+		conditions.WithConditions(
+			clusterv1.ReadyCondition,
```
Comment (smira) [inline controllers/taloscontrolplane_controller.go:797]: I don't see where we're setting this one?

File: controllers/taloscontrolplane_controller.go [hunk 4]
```diff
@@ -736,77 +790,30 @@ func (r *TalosControlPlaneReconciler) addClusterOwnerToObj(ctx context.Context,
...
+			controlplanev1.ResizedCondition,
+			controlplanev1.MachinesReadyCondition,
+			controlplanev1.AvailableCondition,
```
Comment (smira) [inline controllers/taloscontrolplane_controller.go:800]: should we have all conditions here?

File: api/v1alpha3/conditions.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	// EtcdClusterHealthyCondition documents the overall etcd cluster's health.
+	EtcdClusterHealthyCondition clusterv1.ConditionType = "EtcdClusterHealthyCondition"
+
+	// EtcdClusterUnhealthyReason (Severity=Error) is set when the etcd cluster is unhealthy.
+	EtcdClusterUnhealthyReason = "EtcdClusterUnhealthy"
+
+	// MachinesCreatedCondition documents that the machines controlled by the TalosControlPlane are created.
+	// When this condition is false, it indicates that there was an error when cloning the infrastructure/bootstrap template or
+	// when generating the machine object.
+	MachinesCreatedCondition clusterv1.ConditionType = "MachinesCreated"
```
Comment (smira) [inline api/v1alpha3/conditions.go:62]: nit: should be break `const()` block here like we do about for a new condition?

File: .drone.yml (hunk 1 above)
Comment (Unix4ever) [reply .drone.yml:8]: I think it wasn't working without privileged.

File: controllers/taloscontrolplane_controller.go (hunk 4 above)
Comment (Unix4ever) [reply controllers/taloscontrolplane_controller.go:800]: I've used this as a reference. Though I forgot to add couple conditions.

File: .drone.yml (hunk 1 above)
Comment (Unix4ever) [reply .drone.yml]: yep, I remember correctly:

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
--
127 | destroying deployed cluster
128 | Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
129 | make: *** [Makefile:120: integration-test] Error 1
```

**Review by smira — APPROVED** [review]

**Pull request merged**